arch/arm/soc/arm/mps2/soc_devices.h

```diff
@@ -10,6 +10,7 @@
 #include <soc_memory_map.h>
 #include <soc_irq.h>
 
+#if !defined(CONFIG_HAS_DTS)
```

Comment:
I was thinking to keep both the options, but if the indication is device tree only, I can remove !CONFIG_HAS_DTS bits.